src/attestation_types/verify/cert_chain.rs

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use openssl::{stack::Stack, x509::*};
+use std::error::Error;
+
+use super::error::VerifyError;
+
+/// This constructs a complete certificate chain by uniting the tenant's chain (from Intel)
+/// with the leaf cert embedded in the platform's Quote.
+#[derive(Clone)]
+pub struct CertChain {
+    chain: Vec<X509>,
+    leaf: X509,
+    max_len: usize,
+}
+
+#[allow(dead_code)]
+impl CertChain {
+    /// The CertChain is constructed from a vector of X509 certificates and
+    /// a leaf certificate. The maximum chain length defaults to 10.
+    pub fn new_from_chain(c: Vec<X509>, leaf: &X509) -> Self {
+        CertChain {
+            chain: c,
+            leaf: leaf.clone(),
+            max_len: 10,
+        }
+    }
+
+    /// The maximum chain length can be set manually.
+    pub fn set_max_len(&mut self, len: usize) {
+        self.max_len = len;
+    }
+
+    /// Returns length of chain, including leaf cert
+    pub fn len(&self) -> usize {
+        self.chain.len() + 1
+    }
+
+    /// This checks that the CertChain's length is under the maximum allowed.
+    pub fn len_ok(&self) -> Result<(), Box<dyn Error>> {
+        if self.chain.len() > self.max_len {
+            panic!("Certificate chain length exceeds max allowable.");
+        }
+        Ok(())
+    }
```

Comment:
Ah, you're right. I think this should error out because of the kind of check it's meant to be (something we decided on last year about a max allowable chain length), but it shouldn't panic.